tests/sentry/api/test_base.py

```diff
@@ -149,3 +151,31 @@ def test_non_json_content_type(self):
         Endpoint().load_json_body(self.request)
 
         assert not self.request.json_body
+
+
+class ServerComponentModeTest(APITestCase):
+    def _test_active_on(self, endpoint_mode, active_mode, expect_to_be_active):
+        @ApiAvailableOn(endpoint_mode)
+        class DecoratedEndpoint(DummyEndpoint):
+            pass
+
+        class EndpointWithDecoratedMethod(DummyEndpoint):
+            @ApiAvailableOn(endpoint_mode)
+            def get(self, request):
+                return super().get(request)
+
+        with override_settings(SERVER_COMPONENT_MODE=active_mode):
+            for endpoint_class in (DecoratedEndpoint, EndpointWithDecoratedMethod):
+                view = endpoint_class.as_view()
+                request = self.make_request(method="GET")
+                response = view(request)
+                assert response.status_code == (200 if expect_to_be_active else 404)
+
+    def test_with_active_mode(self):
+        self._test_active_on(ServerComponentMode.CUSTOMER, ServerComponentMode.CUSTOMER, True)
```

Comment:
Is it worthwhile having a test for control == control?

Reply:
Sure, I can add that.